drivers/SmartThings/matter-appliance/src/matter-dishwasher/init.lua

```diff
@@ -88,6 +126,62 @@ local function supported_temperature_levels_attr_handler(driver, device, ib, res
   device:emit_event_for_endpoint(ib.endpoint_id, event)
 end
 
+local function temperature_setpoint_attr_handler(driver, device, ib, response)
+  local tn_eps = embedded_cluster_utils.get_endpoints(device, clusters.TemperatureControl.ID,
+    { feature_bitmap = clusters.TemperatureControl.types.Feature.TEMPERATURE_NUMBER })
+  if #tn_eps == 0 then
+    device.log.warn(string.format("Device does not support TEMPERATURE_NUMBER feature"))
+    return
+  end
+  device.log.info(string.format("temperature_setpoint_attr_handler: %d", ib.data.value))
+
+  local min_field = string.format("%s-%d", setpoint_limit_device_field.MIN_TEMP, ib.endpoint_id)
+  local max_field = string.format("%s-%d", setpoint_limit_device_field.MAX_TEMP, ib.endpoint_id)
+  local min = device:get_field(min_field) or DISHWASHER_MIN_TEMP_IN_C
+  local max = device:get_field(max_field) or DISHWASHER_MAX_TEMP_IN_C
+  local temp = ib.data.value / 100.0
+  local unit = "C"
+  local range = {
+    minimum = min,
+    maximum = max,
+  }
+
+  -- Only emit the capability for RPC version >= 5, since unit conversion for
+  -- range capabilities is only supported in that case.
+  if version.rpc >= 5 then
+    device:emit_event_for_endpoint(ib.endpoint_id,
+      capabilities.temperatureSetpoint.temperatureSetpointRange({value = range, unit = unit}, {visibility = {displayed = false}}))
+  end
+
+  device:emit_event_for_endpoint(ib.endpoint_id,
+    capabilities.temperatureSetpoint.temperatureSetpoint({value = temp, unit = unit}))
+end
+
+local function setpoint_limit_handler(limit_field)
+  return function(driver, device, ib, response)
+    local tn_eps = embedded_cluster_utils.get_endpoints(device, clusters.TemperatureControl.ID,
+      { feature_bitmap = clusters.TemperatureControl.types.Feature.TEMPERATURE_NUMBER })
+    if #tn_eps == 0 then
+      device.log.warn(string.format("Device does not support TEMPERATURE_NUMBER feature"))
+      return
+    end
+    local field = string.format("%s-%d", limit_field, ib.endpoint_id)
+    local val = ib.data.value / 100.0
+
+    -- We clamp the max and min values as per the assumed oven temperature ranges.
+    if val < DISHWASHER_MIN_TEMP_IN_C or val > DISHWASHER_MAX_TEMP_IN_C then
```

Comment:
I think you could simplify this by using `utils.clamp_value()` here and in the other places where this logic is used. Example from matter-switch [here](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/cf148fe773ea89daa85584e33ebffcf1dbd5e543/drivers/SmartThings/matter-switch/src/init.lua#L311).

Reply:
Great suggestion! I updated it in each driver that used this logic in the latest commit